### DISCUSSION TITLE
Question about T6 Pro second stage cooling mode reported status

### DISCUSSION THREAD

**Discussion by brianmichalk1** [zwave-js/zwave-js-ui#3723] [Q&A] [answered]
I have 13 HVAC units, six of which have two stage compressors.  All controlled by T6 thermostats over ZWave.  I have configured the tstat for two stage operation, and I can see in the temperature data what looks like throttling of the second compressor when temps get close to the setpoint.
In the status messages, I can see transition from status=0 to status=2
z-ui_current.log:2024-05-22 07:05:12.715 INFO Z-WAVE: [Node 085] Value updated: 66-0-state 0 => 2
z-ui_current.log:2024-05-22 07:08:17.968 INFO Z-WAVE: [Node 017] Value updated: 66-0-state 2 => 0
z-ui_current.log:2024-05-22 07:11:52.710 INFO Z-WAVE: [Node 085] Value updated: 66-0-state 2 => 0
z-ui_current.log:2024-05-22 07:29:52.697 INFO Z-WAVE: [Node 085] Value updated: 66-0-state 0 => 2
z-ui_current.log:2024-05-22 07:45:05.686 INFO Z-WAVE: [Node 085] Value updated: 66-0-state 2 => 2
z-ui_current.log:2024-05-22 07:50:45.683 INFO Z-WAVE: [Node 085] Value updated: 66-0-state 2 => 0
z-ui_current.log:2024-05-22 08:27:15.665 INFO Z-WAVE: [Node 085] Value updated: 66-0-state 0 => 2
At 7:30, both compressors come on when temp setpoint is changed, and I can see the response in the slope of the temperature status.  About 15 minutes later, compressor2 is turned off, but there's no difference in the state, which according to the documentation, should be a 9 => 2.
I wonder if that 2=>2 message is the event I'm looking for?
I have looked through the code here but don't see where the zwave info is placed into the MQTT packet.  I can't tell if the problem is here or in the T6Pro firmware.

      "states": [
        {
          "text": "Idle",
          "value": 0
        },
        {
          "text": "Heating",
          "value": 1
        },
        {
          "text": "Cooling",
          "value": 2
        },
        {
          "text": "Fan Only",
          "value": 3
        },
        {
          "text": "Pending Heat",
          "value": 4
        },
        {
          "text": "Pending Cool",
          "value": 5
        },
        {
          "text": "Vent/Economizer",
          "value": 6
        },
        {
          "text": "Aux Heating",
          "value": 7
        },
        {
          "text": "2nd Stage Heating",
          "value": 8
        },
        {
          "text": "2nd Stage Cooling",
          "value": 9
        },
        {
          "text": "2nd Stage Aux Heat",
          "value": 10
        },
        {
          "text": "3rd Stage Aux Heat",
          "value": 11
        }

**Comment by robertsLando** [comment]
> I have looked through the code here but don't see where the zwave info is placed into the MQTT packet.

You should see a json with value 2 in the payload of the valueid 66-0-state od node 85. That would mean (based on the states above) that the state is on cooling

@AlCalzone any clue?

**Reply by AlCalzone** [reply, marked as answer]
These reports come from the device, so if it reports Cooling (2) instead of 2nd Stage Cooling (9), there's no way we can know. According to your description, I assume the report at 07:29:52.697 should have been a 9, not a 2.

So yes, firmware issue.

**Reply by brianmichalk1** [reply]
Thanks.  I'll see if I can find someone at Honeywell to notify.
